presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -562,6 +583,26 @@ private AmazonS3Client createAmazonS3Client(URI uri, Configuration hadoopConfig,
         return client;
     }
 
+    private EncryptionMaterialsProvider createEncryptionMaterialsProvider(Configuration hadoopConfig)
+    {
+        EncryptionMaterialsProvider emp = null;
+        String empClassName = hadoopConfig.get(S3_ENCRYPTION_MATERIALS_PROVIDER);
```

Comment:
OK – I didn't know about HiveClientConfig and HiveClientConfigurationUpdater, I'll make changes there to add the encryption materials provider classname.  I would still pass the hadoop config into the EMP implementation though (if it implements Configurable), as the specific configs for a given EMP will be different for each implementation.  Does that work?

Thanks,


From: David Phillips <notifications@github.com<mailto:notifications@github.com>>
Reply-To: facebook/presto <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Tuesday, October 27, 2015 at 10:19 AM
To: facebook/presto <presto@noreply.github.com<mailto:presto@noreply.github.com>>
Cc: Nate Sammons <nate.sammons@nasdaq.com<mailto:nate.sammons@nasdaq.com>>
Subject: Re: [presto] Add support for S3 EncryptionMaterialsProvider to PrestoS3FileSystem (#3802)


In presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java<https://github.com/facebook/presto/pull/3802#discussion_r43145403>:

> @@ -562,6 +583,26 @@ private AmazonS3Client createAmazonS3Client(URI uri, Configuration hadoopConfig,
>          return client;
>      }
>
> +    private EncryptionMaterialsProvider createEncryptionMaterialsProvider(Configuration hadoopConfig)
> +    {
> +        EncryptionMaterialsProvider emp = null;
> +        String empClassName = hadoopConfig.get(S3_ENCRYPTION_MATERIALS_PROVIDER);


Configuring this via the Hadoop config system doesn't fit with the way Presto and the Hive connector do configuration. We have first class config support in HiveClientConfig for most everything that a user would want to configure. All of these S3_ Hadoop config variables are for passing the Hive connector's configuration into PrestoS3FileSystem (because it is created by Hadoop and thus we don't have direct access to configure it). See HdfsConfigurationUpdater for where these are propagated from HiveClientConfig into the Hadoop Configuration object.

—
Reply to this email directly or view it on GitHub<https://github.com/facebook/presto/pull/3802/files#r43145403>.
***********************************************************
CONFIDENTIALITY NOTICE: This e-mail and any attachments are for the exclusive and confidential use of the intended recipient and may constitute non-public information. If you received this e-mail in error, disclosing, copying, distributing or taking any action in reliance of this e-mail is strictly prohibited and may be unlawful. Instead, please notify us immediately by return e-mail and promptly delete this message and its attachments from your computer system. We do not waive any work product or other applicable legal privilege(s) by the transmission of this message.
***********************************************************